mist-common/src/main/java/edu/snu/mist/common/operators/ApplyStatefulOperator.java

```diff
@@ -86,16 +95,37 @@ public void processLeftWatermark(final MistWatermarkEvent input) {
     outputEmitter.emitWatermark(input);
     if (input.isCheckpoint()) {
       latestCheckpointTimestamp = input.getTimestamp();
+      final Map<String, Object> stateMap = new HashMap<>();
+      stateMap.put("applyStatefulFunctionState", applyStatefulFunction.getCurrentState());
```

Comment:
The state should be deep copied if we want to store the state at this time.

Reply:
This is true. I will fix this.